ZenPacks/zenoss/OpenStackInfrastructure/poll_openstack.py

```diff
@@ -464,6 +423,18 @@ def printJSON(self):
         data = None
         try:
             data = yield self.getData()
+
+            if self.api_error_messages:
+                summary = 'APIClient Errors: '
+                summary += '. '.join(self.api_error_messages)
+
+                data['events'].append(dict(
+                    severity=3,
+                    summary=summary,
+                    eventKey='openStackFailure',
+                    eventClassKey='openStackApiClientStatus',
+                ))
+
```

Comment:
I would suggest using the existing openStackFailure event for this.  I'm not sure what the difference really is, otherwise.

If you do think there's a difference here and want both, then you need to add a clear (severity 0) event for when there are no api errors, otherwise these events won't clear on their own.

Reply:
We already use the openStackFailure here for eventClass, which I believe is used to clear the event just below. The eventClassKey='openStackApiClientStatus' is used to allow the user to have class key mapping control. Ive added an extra mapping entry to allow for this.